src/providers/exchange.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { createContext, useState, useContext, useEffect } from 'react';
+import { ethers, BigNumber, constants, Contract } from 'ethers';
+import { chainlinkAbi } from '../abis/chainlink';
+import { useNetworkProvider } from './ethers';
+import { useSafeAppsSDK } from '@gnosis.pm/safe-apps-react-sdk';
+import { getLootboxERC20Sources, getExternalErc20Sources } from '../utils/getERC20Sources';
+import { erc20Source, usePoolData } from './pools';
+import { flatten } from '../utils/flatten';
+import { GetTokenExchange } from '../ptGraphClient';
+const CHAINLINK_ADDRESSES = {
+  [1]: '0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419',
+  [4]: '0x8A753747A1Fa494EC906cE90E9f37563A8AF630e',
+};
+
+type ExchangeDict = {
+  [key: string]: {
+    derivedEth: string;
+    decimals: string;
+  };
+};
+
+type InitalizedExchangeDict = {
+  initialized: boolean;
+  exchangeDict: ExchangeDict;
+};
+
+type ExchangeType = {
+  ethToUsd: BigNumber;
+  tokenExchange: InitalizedExchangeDict;
+};
+
+// EIGHT DECIMALS
+
+const ExchangeContext = createContext<ExchangeType>({
+  ethToUsd: constants.Zero,
+  tokenExchange: {
+    initialized: false,
+    exchangeDict: {},
```

Comment:
pull in eth to usd from chainlink, other tokens to eth from uni

Reply:
think pool together might be using a different eth to usd oracle which would explain why the prize values we get are slightly different.

Reply:
This is great, I would say it's probably fine to use off chain api's etc for this kind of data as the chainlink oracle only updates periodically or if there is a 0.5% movement. 
This is 'always free' though so great approach!
https://data.chain.link/eth-usd

Reply:
In the future if its easier*, otherwise this is fantastic!!